src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -132,13 +132,6 @@ private BoundPattern BindDiscardPattern(DiscardPatternSyntax node, TypeSymbol in
             bool hasErrors,
             DiagnosticBag diagnostics)
         {
-            SyntaxNode innerExpression = node.Expression.SkipParens();
-            if (innerExpression.Kind() == SyntaxKind.DefaultLiteralExpression)
-            {
-                diagnostics.Add(ErrorCode.ERR_DefaultPattern, innerExpression.Location);
```

Comment:
similarly: is 'default' now a valid constant pattern?

Reply:
that scenario now produces: "no target type for `default`"

---
In reply to: [313659852](https://github.com/dotnet/roslyn/pull/37596#discussion_r313659852) [](ancestors = 313659852)